Doc/reference/datamodel.rst

```diff
@@ -2108,7 +2108,7 @@ Emulating callable objects
    .. index:: pair: call; instance
 
    Called when the instance is "called" as a function; if this method is defined,
-   ``x(arg1, arg2, ...)`` is a shorthand for ``x.__call__(arg1, arg2, ...)``.
+   the class receives the instance and all arguments of the call.
```

Comment:
I know what is supposed to be conveyed here, and I find this more confusing than the original. Diving into the internals of how calls to instance methods work is detracting from the discussion of what __call__ does.

Reply:
This current version don't work for me, "if this method is defined, the class receives the instance and all arguments of the call." looks wrong, It's not the class that receives the instance and all arguments.

I'd propose something like:

```Called when the instance is "called" as a function. Receives the instance and the function parameters.```

If you want to almost keep the example you could add something like:

```So ``obj(arg1, arg2, ...)`` calls ``__call__(arg1, arg2, ...)`` on ``obj``.```

Reply:
I don’t think that last suggestion (from Julien) works (call on obj?).

`type(x).__call__(x, arg1, ...)` is closer to what happens, and the call matches the way the method is defined (with a self parameter).

Reply:
@merwok I agree. Made a change based on your suggestion, what do you think?

Reply:
@vstinner You said «I would prefer to [not] use type() in the doc, since technically the builtin function can be overriden» → that’s what we say «roughly»!  Is the latest version good for you or you would prefer `obj.__class__.__call__(obj, etc)`?

@csabella I’m not sure I understand your review; could you comment on the latest version?  Thanks!